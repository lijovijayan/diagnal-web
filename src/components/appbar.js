import React, { useCallback, useState } from 'react'
import { debounce } from 'lodash'
const searchIcon = `${process.env.PUBLIC_URL}/images/search.png`
const backIcon = `${process.env.PUBLIC_URL}/images/Back.png`

export function AppBar({ onChange, title }) {
    const [showSearch, showSearchInput] = useState(false)
    const handler = useCallback(debounce(onChange, 400), [])

    function onSwitchSearchIcon() {
        if (showSearch) {
            onChange('')
        }
        showSearchInput(!showSearch)
    }

    return (
        <div
            className={`search-bar flex flex-row items-center ${
                showSearch ? 'show-search' : ''
            }`}
        >
            <img
                className={'back-icon ml-3'}
                src={backIcon}
                alt={'search'}
                draggable={false}
            />
            <div className="flex-1 text-xl text-left flex flex-row justify-between items-center">
                <span className="pl-3 header-title">{title}</span>
                {showSearch && (
                    <input
                        type="text"
                        class="text-base search-input mx-3 px-3 flex-1"
                        onChange={(e) => handler(e.target.value)}
                        placeholder="search here..."
                    />
                )}
            </div>
            <img
                id="searchIcon"
                className={'search-icon mr-3'}
                src={searchIcon}
                alt={'search'}
                draggable={false}
                onClick={onSwitchSearchIcon}
            />
        </div>
    )
}
