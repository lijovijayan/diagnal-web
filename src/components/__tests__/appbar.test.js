import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AppBar } from '../appbar'

describe('testing appbar', () => {
    it('rendered properly', () => {
        const appbar = <AppBar onChange={() => {}} title="Appbar Render Test" />
        const componentWrapper = shallow(appbar)
        let componentTree = toJson(componentWrapper)
        expect(componentTree).toMatchSnapshot()

        componentWrapper.find('#searchIcon').simulate('click')
        componentTree = toJson(componentWrapper)
        expect(componentTree).toMatchSnapshot()

        componentWrapper.find('#searchIcon').simulate('click')
        componentTree = toJson(componentWrapper)
        expect(componentTree).toMatchSnapshot()
    })
})
