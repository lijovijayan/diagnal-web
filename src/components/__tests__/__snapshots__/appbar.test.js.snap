// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing appbar rendered properly 1`] = `
<div
  className="search-bar flex flex-row items-center "
>
  <img
    alt="search"
    className="back-icon ml-3"
    draggable={false}
    src="/images/Back.png"
  />
  <div
    className="flex-1 text-xl text-left flex flex-row justify-between items-center"
  >
    <span
      className="pl-3 header-title"
    >
      Appbar Render Test
    </span>
  </div>
  <img
    alt="search"
    className="search-icon mr-3"
    draggable={false}
    id="searchIcon"
    onClick={[Function]}
    src="/images/search.png"
  />
</div>
`;

exports[`testing appbar rendered properly 2`] = `
<div
  className="search-bar flex flex-row items-center show-search"
>
  <img
    alt="search"
    className="back-icon ml-3"
    draggable={false}
    src="/images/Back.png"
  />
  <div
    className="flex-1 text-xl text-left flex flex-row justify-between items-center"
  >
    <span
      className="pl-3 header-title"
    >
      Appbar Render Test
    </span>
    <input
      class="text-base search-input mx-3 px-3 flex-1"
      onChange={[Function]}
      placeholder="search here..."
      type="text"
    />
  </div>
  <img
    alt="search"
    className="search-icon mr-3"
    draggable={false}
    id="searchIcon"
    onClick={[Function]}
    src="/images/search.png"
  />
</div>
`;

exports[`testing appbar rendered properly 3`] = `
<div
  className="search-bar flex flex-row items-center "
>
  <img
    alt="search"
    className="back-icon ml-3"
    draggable={false}
    src="/images/Back.png"
  />
  <div
    className="flex-1 text-xl text-left flex flex-row justify-between items-center"
  >
    <span
      className="pl-3 header-title"
    >
      Appbar Render Test
    </span>
  </div>
  <img
    alt="search"
    className="search-icon mr-3"
    draggable={false}
    id="searchIcon"
    onClick={[Function]}
    src="/images/search.png"
  />
</div>
`;
